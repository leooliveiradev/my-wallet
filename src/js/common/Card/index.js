import React, { memo } from 'react';
import styled from 'styled-components';
import { fromTheme } from 'app/js/utils';

export const sizes = {
  md: '30rem',
};

export const CardWrapper = styled.div`
  width: ${({ size }) => sizes[size]};
  padding: 2rem;
  box-shadow: ${({ withShadow }) => (withShadow ? '2px 4px 8px 0 rgba(46, 61, 73, 0.2)' : 'none')};
  font-family: ${fromTheme('fontStyle.primary')};
  font-size: ${fromTheme('fontSize.small')};
  color: ${fromTheme('color.text')};
  background: ${fromTheme('color.white')};
`;

export const CardContent = styled.div`
  display: flex;
`;

export const CardTitle = styled.h2`
  color: ${fromTheme('color.primary')};
  font-weight: ${fromTheme('fontWeight.semibold')};
  font-size: 1.4rem;
`;

const Card = memo(({
  size, title, children, withShadow,
}) => (
  <CardWrapper
    size={size}
    withShadow={withShadow}
  >
    <CardTitle>
      {title}
    </CardTitle>
    <CardContent>
      {children}
    </CardContent>
  </CardWrapper>
));

export default Card;
