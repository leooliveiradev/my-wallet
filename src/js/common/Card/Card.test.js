import React from 'react';
import Card from 'app/js/common/Card';
import renderer from 'react-test-renderer';

describe('Card snapshot', () => {
  it('should render the Card', () => {
    const component = renderer.create(
      <Card
        title="title of transaction"
        size="md"
      >
        <p>
          Card body
        </p>
      </Card>,
    ).toJSON();
    expect(component).toMatchSnapshot();
  });
});
