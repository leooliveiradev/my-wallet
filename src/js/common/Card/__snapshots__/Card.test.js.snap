// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card snapshot should render the Card 1`] = `
<div
  className="sc-bdVaJa jtTKft"
  size="md"
>
  <h2
    className="sc-htpNat jzBuPf"
  >
    title of transaction
  </h2>
  <div
    className="sc-bwzfXH jIEWCi"
  >
    <p>
      Card body
    </p>
  </div>
</div>
`;
